### ISSUE TITLE
"Could not find nokogiri-1.7.0.1 in any of the sources"

### ISSUE THREAD

**Issue by ryan-sattler** [bundler/bundler#6006]
This issue seems to be the same problem as https://github.com/bundler/bundler/issues/5046 , but that one has already been marked as closed and I have been unable to solve it.

Details:
When trying to run `bundle exec rake start` (or other uses of `bundle exec`, or just`rake start`) in a Rails project, I get this error:
```
Could not find nokogiri-1.7.0.1 in any of the sources
Run `bundle install` to install missing gems.
```

However, `bundle install` seemed to have worked correctly, including installing Nokogiri successfully. Using `gem` to manually uninstall and reinstall Nokogiri did not help.

This issue happens with both bundler 1.14.4 and 1.16.0.pre.1, and for various versions of Nokogiri (1.7.0.1, 1.7.2, 1.8.0).

Here's my env:

## Environment

```
Bundler   1.16.0.pre.1
RubyGems  2.6.11
Ruby      2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
GEM_HOME  /Users/ryan.sattler/.gem/ruby/2.4.1
GEM_PATH  /Users/ryan.sattler/.gem/ruby/2.4.1:/Users/ryan.sattler/.rubies/ruby-2.4.1/lib/ruby/gems/2.4.0
Git       2.13.4
Platform  x86_64-darwin-16
OpenSSL   OpenSSL 1.0.2k  26 Jan 2017
```

## Bundler Build Metadata

```
Built At          2017-09-05
Git SHA           5ffc8b010
Released Version  true
```

### Gemfile

```ruby
source 'https://rubygems.org'
# NOTE: All gems for production code should have version specified.
#       Big libraries use patch version, e.g. '~> 1.2.0'
#       Smaller libraries use minor version, e.g. '~> 3.4'
#       Development/test libraries we just use the latest

gem 'rails', '~> 5.0'

# Archie DB and EAI integration
gem 'sequel', '~> 4.43'
gem 'tiny_tds', '~> 1.1'
gem 'rest-client', '~> 2.0'

# Data libraries
gem 'multi_json', '~> 1.12'
gem 'oj', '2.18'
gem 'nokogiri', '1.7.0.1'
gem 'roxml', '~> 3.3'
gem 'roar-rails', '~> 1.0'

# Capistrano for deployment
gem 'capistrano', '~> 3.7'
gem 'capistrano-rbenv', '~> 2.1'
gem 'capistrano-bundler', '~> 1.2'

# Unicorn as the app server
gem 'unicorn', '~> 5.2'
gem 'unicorn-worker-killer', '~> 0.4'

gem 'newrelic_rpm'

# Development and Testing tools
# These we can just keep current
group :development, :test do
  gem 'dotenv-rails', '~> 2.2'

  gem 'rubocop', '~> 0.47'

  gem 'rspec-mocks', '~> 3.5'
  gem 'rspec-rails', '~> 3.5'
  gem 'rspec-collection_matchers', '~> 1.1'
  gem 'faker', '~> 1.7'
  gem 'shoulda-matchers', '~> 3.1', require: false
  gem 'webmock', '~> 2.3'

  gem 'pact', '~> 1.7.0'
  gem 'vcr', '~> 3.0'

  gem 'simplecov', '~> 0.13'
  gem 'codeclimate-test-reporter', '~> 0.6', require: nil

  gem 'http_server_manager', '~> 0.5'

  gem 'awesome_print'
  gem 'pry'
end
```

### Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.1.3)
      actionpack (= 5.1.3)
      nio4r (~> 2.0)
      websocket-driver (~> 0.6.1)
    actionmailer (5.1.3)
      actionpack (= 5.1.3)
      actionview (= 5.1.3)
      activejob (= 5.1.3)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.1.3)
      actionview (= 5.1.3)
      activesupport (= 5.1.3)
      rack (~> 2.0)
      rack-test (~> 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.1.3)
      activesupport (= 5.1.3)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (5.1.3)
      activesupport (= 5.1.3)
      globalid (>= 0.3.6)
    activemodel (5.1.3)
      activesupport (= 5.1.3)
    activerecord (5.1.3)
      activemodel (= 5.1.3)
      activesupport (= 5.1.3)
      arel (~> 8.0)
    activesupport (5.1.3)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (~> 0.7)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    addressable (2.5.2)
      public_suffix (>= 2.0.2, < 4.0)
    airbrussh (1.3.0)
      sshkit (>= 1.6.1, != 1.7.0)
    arel (8.0.0)
    ast (2.3.0)
    awesome_print (1.8.0)
    builder (3.2.3)
    capistrano (3.9.0)
      airbrussh (>= 1.0.0)
      i18n
      rake (>= 10.0.0)
      sshkit (>= 1.9.0)
    capistrano-bundler (1.2.0)
      capistrano (~> 3.1)
      sshkit (~> 1.2)
    capistrano-rbenv (2.1.1)
      capistrano (~> 3.1)
      sshkit (~> 1.3)
    codeclimate-test-reporter (0.6.0)
      simplecov (>= 0.7.1, < 1.0.0)
    coderay (1.1.2)
    concurrent-ruby (1.0.5)
    crack (0.4.3)
      safe_yaml (~> 1.0.0)
    declarative (0.0.9)
    declarative-option (0.1.0)
    diff-lcs (1.3)
    diffy (3.2.0)
    docile (1.1.5)
    domain_name (0.5.20170404)
      unf (>= 0.0.5, < 1.0.0)
    dotenv (2.2.1)
    dotenv-rails (2.2.1)
      dotenv (= 2.2.1)
      railties (>= 3.2, < 5.2)
    erubi (1.6.1)
    faker (1.8.4)
      i18n (~> 0.5)
    find_a_port (1.0.1)
    get_process_mem (0.2.1)
    globalid (0.4.0)
      activesupport (>= 4.2.0)
    hashdiff (0.3.6)
    http-cookie (1.0.3)
      domain_name (~> 0.5)
    http_server_manager (0.5.0)
      rake (>= 10.4)
      sys-proctree (~> 0.0)
      wait_until (~> 0.3)
    i18n (0.8.6)
    json (2.1.0)
    kgio (2.11.0)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.6)
      mime-types (>= 1.16, < 4)
    method_source (0.8.2)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mini_portile2 (2.1.0)
    minitest (5.10.3)
    multi_json (1.12.2)
    net-scp (1.2.1)
      net-ssh (>= 2.6.5)
    net-ssh (4.1.0)
    netrc (0.11.0)
    newrelic_rpm (4.4.0.336)
    nio4r (2.1.0)
    nokogiri (1.7.0.1)
      mini_portile2 (~> 2.1.0)
    oj (2.18.0)
    pact (1.7.0)
      awesome_print (~> 1.1)
      find_a_port (~> 1.0.1)
      json
      pact-mock_service (~> 0.5.0)
      pact-support (~> 0.4.0)
      rack-test (~> 0.6.2)
      randexp (~> 0.1.7)
      rspec (>= 2.14)
      term-ansicolor (~> 1.0)
      thor
      webrick
    pact-mock_service (0.5.5)
      find_a_port (~> 1.0.1)
      json
      pact-support (~> 0.4.0)
      rack
      rack-test (~> 0.6.2)
      rspec (>= 2.14)
      term-ansicolor (~> 1.0)
      thor
      webrick
    pact-support (0.4.4)
      awesome_print (~> 1.1)
      find_a_port (~> 1.0.1)
      json
      rack-test (~> 0.6.2)
      randexp (~> 0.1.7)
      rspec (>= 2.14)
      term-ansicolor (~> 1.0)
      thor
    parallel (1.12.0)
    parser (2.4.0.0)
      ast (~> 2.2)
    powerpack (0.1.1)
    pry (0.10.4)
      coderay (~> 1.1.0)
      method_source (~> 0.8.1)
      slop (~> 3.4)
    public_suffix (3.0.0)
    rack (2.0.3)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (5.1.3)
      actioncable (= 5.1.3)
      actionmailer (= 5.1.3)
      actionpack (= 5.1.3)
      actionview (= 5.1.3)
      activejob (= 5.1.3)
      activemodel (= 5.1.3)
      activerecord (= 5.1.3)
      activesupport (= 5.1.3)
      bundler (>= 1.3.0)
      railties (= 5.1.3)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    railties (5.1.3)
      actionpack (= 5.1.3)
      activesupport (= 5.1.3)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rainbow (2.2.2)
      rake
    raindrops (0.19.0)
    rake (12.0.0)
    randexp (0.1.7)
    representable (3.0.4)
      declarative (< 0.1.0)
      declarative-option (< 0.2.0)
      uber (< 0.2.0)
    responders (2.4.0)
      actionpack (>= 4.2.0, < 5.3)
      railties (>= 4.2.0, < 5.3)
    rest-client (2.0.2)
      http-cookie (>= 1.0.2, < 2.0)
      mime-types (>= 1.16, < 4.0)
      netrc (~> 0.8)
    roar (1.1.0)
      representable (~> 3.0.0)
    roar-rails (1.1.0)
      actionpack
      railties (>= 3.0.0)
      responders
      roar (~> 1.1.0)
      test_xml (>= 0.1.6)
      uber (< 0.2.0)
    roxml (3.3.1)
      activesupport (>= 2.3.0)
      nokogiri (>= 1.3.3)
    rspec (3.6.0)
      rspec-core (~> 3.6.0)
      rspec-expectations (~> 3.6.0)
      rspec-mocks (~> 3.6.0)
    rspec-collection_matchers (1.1.3)
      rspec-expectations (>= 2.99.0.beta1)
    rspec-core (3.6.0)
      rspec-support (~> 3.6.0)
    rspec-expectations (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-mocks (3.6.0)
      diff-lcs (>= 1.2.0, < 2.0)
      rspec-support (~> 3.6.0)
    rspec-rails (3.6.1)
      actionpack (>= 3.0)
      activesupport (>= 3.0)
      railties (>= 3.0)
      rspec-core (~> 3.6.0)
      rspec-expectations (~> 3.6.0)
      rspec-mocks (~> 3.6.0)
      rspec-support (~> 3.6.0)
    rspec-support (3.6.0)
    rubocop (0.49.1)
      parallel (~> 1.10)
      parser (>= 2.3.3.1, < 3.0)
      powerpack (~> 0.1)
      rainbow (>= 1.99.1, < 3.0)
      ruby-progressbar (~> 1.7)
      unicode-display_width (~> 1.0, >= 1.0.1)
    ruby-progressbar (1.8.1)
    safe_yaml (1.0.4)
    sequel (4.49.0)
    shoulda-matchers (3.1.2)
      activesupport (>= 4.0.0)
    simplecov (0.15.0)
      docile (~> 1.1.0)
      json (>= 1.8, < 3)
      simplecov-html (~> 0.10.0)
    simplecov-html (0.10.2)
    slop (3.6.0)
    sprockets (3.7.1)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    sshkit (1.14.0)
      net-scp (>= 1.1.2)
      net-ssh (>= 2.8.0)
    sys-proctable (1.0.0-universal-aix5)
    sys-proctree (0.0.10)
      sys-proctable (= 1.0.0)
    term-ansicolor (1.6.0)
      tins (~> 1.0)
    test_xml (0.1.8)
      diffy (~> 3.0)
      nokogiri (>= 1.3.2)
    thor (0.20.0)
    thread_safe (0.3.6)
    tins (1.15.0)
    tiny_tds (1.3.0)
      mini_portile2 (~> 2.0)
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    uber (0.1.0)
    unf (0.1.4)
      unf_ext
    unf_ext (0.0.7.4)
    unicode-display_width (1.3.0)
    unicorn (5.3.0)
      kgio (~> 2.6)
      raindrops (~> 0.7)
    unicorn-worker-killer (0.4.4)
      get_process_mem (~> 0)
      unicorn (>= 4, < 6)
    vcr (3.0.3)
    wait_until (0.3.0)
    webmock (2.3.2)
      addressable (>= 2.3.6)
      crack (>= 0.3.2)
      hashdiff
    webrick (1.3.1)
    websocket-driver (0.6.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.2)

PLATFORMS
  ruby

DEPENDENCIES
  awesome_print
  capistrano (~> 3.7)
  capistrano-bundler (~> 1.2)
  capistrano-rbenv (~> 2.1)
  codeclimate-test-reporter (~> 0.6)
  dotenv-rails (~> 2.2)
  faker (~> 1.7)
  http_server_manager (~> 0.5)
  multi_json (~> 1.12)
  newrelic_rpm
  nokogiri (= 1.7.0.1)
  oj (= 2.18)
  pact (~> 1.7.0)
  pry
  rails (~> 5.0)
  rest-client (~> 2.0)
  roar-rails (~> 1.0)
  roxml (~> 3.3)
  rspec-collection_matchers (~> 1.1)
  rspec-mocks (~> 3.5)
  rspec-rails (~> 3.5)
  rubocop (~> 0.47)
  sequel (~> 4.43)
  shoulda-matchers (~> 3.1)
  simplecov (~> 0.13)
  tiny_tds (~> 1.1)
  unicorn (~> 5.2)
  unicorn-worker-killer (~> 0.4)
  vcr (~> 3.0)
  webmock (~> 2.3)

BUNDLED WITH
   1.16.0.pre.1
```

**Comment by segiddins** [comment]
I'm unable to get `tiny_tds` installed locally to reproduce this

**Comment by arbonap** [comment]
I also couldn't get `tiny_tds` installed locally until I did `brew install freetds`. After brew installing, I was able to successfully `bundle install`.

**Comment by arbonap** [comment]
Hmm, so I'm not able to reproduce this nokogiri error. But, I made an example rails app with @ryan-sattler's Gemfile. After `brew install freetds` and `bundle install`-ing, when I try to run `rails s`, I was getting this stacktrace: 
```
➜  example git:(master) ✗ rails s
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roxml-3.3.1/lib/roxml/definition.rb:156: warning: constant ::Fixnum is deprecated
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:86:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'roar-rails'. (Bundler::GemRequireError)
Gem Load Error is: uninitialized constant ActionController::Responder
Did you mean?  ActionController::Renderer
Backtrace for gem load error is:
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar/rails/rails4_0_strategy.rb:2:in `<module:Rails>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar/rails/rails4_0_strategy.rb:1:in `<top (required)>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar-rails.rb:40:in `require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar-rails.rb:40:in `<module:Rails>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar-rails.rb:20:in `<module:Roar>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/roar-rails-1.1.0/lib/roar-rails.rb:19:in `<top (required)>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:83:in `require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:83:in `block (2 levels) in require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:78:in `each'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:78:in `block in require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:67:in `each'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:67:in `require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler.rb:114:in `require'
/Users/Patricia/workspace/bundler-rails/example/config/application.rb:7:in `<top (required)>'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:129:in `require'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:129:in `block in perform'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:126:in `tap'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:126:in `perform'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/command/base.rb:63:in `perform'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/command.rb:44:in `invoke'
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands.rb:16:in `<top (required)>'
bin/rails:9:in `require'
bin/rails:9:in `<main>'
Bundler Error Backtrace:
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:82:in `block (2 levels) in require'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:78:in `each'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:78:in `block in require'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:67:in `each'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler/runtime.rb:67:in `require'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/bundler-1.16.0.pre.1/lib/bundler.rb:114:in `require'
	from /Users/Patricia/workspace/bundler-rails/example/config/application.rb:7:in `<top (required)>'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:129:in `require'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:129:in `block in perform'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:126:in `tap'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands/server/server_command.rb:126:in `perform'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/command/base.rb:63:in `perform'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/command.rb:44:in `invoke'
	from /Users/Patricia/.rvm/gems/ruby-2.4.1/gems/railties-5.1.3/lib/rails/commands.rb:16:in `<top (required)>'
	from bin/rails:9:in `require'
	from bin/rails:9:in `<main>'
```
I read up on the [`roar-rails`](https://github.com/apotonick/roar-rails) docs that if you're using >= Rails 4.2, you need `gem "responders"` before  `gem "roar-rails" ` in your Gemfile. Since this is Rails 5, I added it in. @ryan-sattler maybe you need to add in the `gem "responders"` to your Gemfile? I `bundle install`-ed again and got further. I tried restarting my server, but then got another stacktrace, mentioning that I needed the `listen` gem:

```
/Users/Patricia/.rvm/gems/ruby-2.4.1/gems/activesupport-5.1.3/lib/active_support/dependencies.rb:292:in `require': 
Could not load the 'listen' gem. Add `gem 'listen'` to the development group of your Gemfile (LoadError)
```
I added `listen` to the Gemfile in the development group section, `bundle install`-ed again, and was able to start the server finally with `bundle exec rails s`. I tried a `rake` command and it worked for me.

**Comment by segiddins** [comment]
This works just fine for me, `bundle exec ruby -e 'p(true)'` succeeds

**Comment by ryan-sattler** [comment]
@arbonap Thanks, but adding those gems didn't make any difference. I still get the could-not-find-Nokogiri error, including from @segiddins example of `bundle exec ruby -e 'p(true)'` which presumably doesn't invoke Rails.

Is it possible something is corrupt and needs to be removed? I tried the various rm statements given in the troubleshooting guide, but they didn't seem to have any effect (gems did not even need to be reinstalled). Is there a way to blow away all the relevant gems/caches/indexes/etc and try again?

**Comment by ryan-sattler** [comment]
Update: After changing bundler to install with a local --path option, I've now got past the nokogiri error. I assume there is something wrong with my global gems. At any rate, I got the same problems as @arbonap, implemented her fixes, and then got stuck on something else, which doesn't seem to be a Bundler issue necessarily. Thanks for you help.

**Comment by segiddins** [comment]
Glad you got things sorted!

**Comment by dm-wyncode** [comment]
I am an instructor at Wyncode Academy. I was recently verifying multiple rails apps created by students.

One student's app would produce the error described above.

The student's repo is here: https://github.com/zmunir93/rails-practice

The commit I used is 04de31717500fdac4927cd4265e8f6a5855faea6

The output of `bundle install` is

```

Fetching gem metadata from https://rubygems.org/.........
Could not find rack-test-0.7.1 in any of the sources

```

Out of around 15 rails apps, his was the only one that produced this error on my environment which is as follows:

ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-linux]

**Comment by ondenman** [comment]
What version of Ruby is he running? I just switched to 2.4.2 and got the same error. Switching back to 2.3.1, I can run bundle install on the same gem file without any complaints.

**Comment by macelii** [comment]
I got the same issue as above and landed here.
Switched to Ruby 2.4.2 and got "Could not find rack" issue.
Went back to 2.3.1, bundle install gave me then "Could not find nokogiri-1.7.0.1 in any of the sources".
Now I am stuck between a rock and a hard place.

**Comment by shaheem007** [comment]
msfconsole
2019-05-06 22:46:19.770 SAST [21222] LOG:  received fast shutdown request
waiting for server to shut down....2019-05-06 22:46:19.775 SAST [21222] LOG:  aborting any active transactions
2019-05-06 22:46:19.780 SAST [21222] LOG:  background worker "logical replication launcher" (PID 21235) exited with exit code 1
2019-05-06 22:46:19.784 SAST [21230] LOG:  shutting down
2019-05-06 22:46:19.942 SAST [21222] LOG:  database system is shut down
 done
server stopped
waiting for server to start....2019-05-06 22:46:20.238 SAST [22670] LOG:  listening on IPv4 address "127.0.0.1", port 5432                                              2019-05-06 22:46:20.253 SAST [22670] LOG:  listening on Unix socket "/data/data/com.termux/files/usr/tmp/.s.PGSQL.5432"
2019-05-06 22:46:20.416 SAST [22674] LOG:  database system was shut down at 2019-05-06 22:46:19 SAST
2019-05-06 22:46:20.444 SAST [22670] LOG:  database system is ready to accept connections
 done
server started
Could not find rake-12.3.2 in any of the sources
Run `bundle install` to install missing gems.

**Comment by shaheem007** [comment]
Could not find rake-12.3.2 in any of the sources
Run `bundle install` to install missing gems.

**Comment by shaheem007** [comment]
Run `bundle install` to install missing gems.
$ bundle install                                        Could not locate Gemfile
$

**Comment by shaheem007** [comment]
Gemfile:
factory_bot_rails was resolved to 5.0.2, which depends
on
    railties was resolved to 4.2.11.1, which depends on
actionpack was resolved to 4.2.11.1, which depends
on
actionview was resolved to 4.2.11.1, which
depends on
rails-dom-testing was resolved to 1.0.9, which
depends on
            nokogiri
dpkg: error processing package metasploit (--configure):
 installed metasploit package post-installation script subprocess returned error exit status 5
Errors were encountered while processing:
 metasploit
E: Sub-process /data/data/com.termux/files/usr/bin/dpkg returned an error code (1)
$

**Comment by shaheem007** [comment]
An error occurred while installing nokogiri
(1.10.2), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.10.2'
--source 'https://rubygems.org/'` succeeds before
bundling.
